include/sparrow/buffer_view.hpp

```diff
@@ -0,0 +1,270 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "sparrow/buffer.hpp"
+
+namespace sparrow
+{
+    /*
+     * @class buffer_view
+     * @brief Object that references but does not own a piece of contiguous memory
+     */
+    template <class T>
+    class buffer_view
```

Comment:
How is this type different from `std::span`?

Reply:
They are very similar indeed. However, `std::span` is missing some APIs in C++20 (like `cbegin` / `cend` ), and we may want to provide additional std;:vector like APIs.

Reply:
Could you add that clarification in the comments, so that we keep a trace in the code?